qiskit/circuit/library/data_preparation/initializer.py

```diff
@@ -86,9 +87,14 @@ def gates_to_uncompute(self) -> QuantumCircuit:
         """Call to create a circuit with gates that take the desired vector to zero.
 
         Returns:
-            Circuit to take ``self.params`` vector to :math:`|{00\\ldots0}\\rangle`
+            QuantumCircuit: circuit to take ``self.params`` vector to :math:`|{00\\ldots0}\\rangle`
         """
-        return self._stateprep._gates_to_uncompute()
+        q = QuantumRegister(self.num_qubits)
+        circuit = QuantumCircuit(q, name="disentangler")
+
+        isom = Isometry(self.params, 0, 0)
```

Comment:
If I recall correctly (maybe we need to double check), `gates_to_uncompute` previously returned a circuit with only `Gate` objects -- now it would return a circuit with an `Instruction`. To fix this we could maybe do 
```python
circuit.compose(isom.definition.inverse(), copy=False, inplace=True)
```
or, the more performant version, just
```python
return isom._gates_to_uncompute()
```
?

Reply:
fixed in [4ae61c7](https://github.com/Qiskit/qiskit/pull/12976/commits/4ae61c78d2d44935076cc0ff84af3cbe731b912b)